src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java

```diff
@@ -69,73 +69,71 @@ protected void onJcrRootElement(String uri, String localName, String qName, Attr
-                                    if (s != null) {
-                                        res = Boolean.valueOf(s);
-                                    }
-                                    return res;
-                                }).toArray();
-                                break;
+                        Object[] values = getValues(property);
+                        if (values.length == 0) {
+                            // ignore empty values (either property.values were empty or value mapping resulted in null 
+                            // results that got filtered)
+                            continue;
                         }
                         if (!property.isMulti) {
-                            // first element to be used in case of singlevalue
-                            value = ((Object[])value)[0];
-                        }
-                         
-                        
-                        if (property.values.length > 0) {
-                            configuration.put(attributeQName, value);
+                            // first element to be used in case of single-value property
+                            configuration.put(attributeQName, values[0]);
+                        } else {
+                            configuration.put(attributeQName, values);
                         }
                     }
                 }
             }
         }
+        
+        private static boolean isEmptyOrNull(@Nullable String s) {
+            return s == null || s.isEmpty();
+        }
+        
+        @NotNull 
+        private static Object[] getValues(@NotNull DocViewProperty property) {
+            Object[] values;
+            switch (property.type) {
+                case PropertyType.DATE:
+                    // Date was never properly supported as osgi configs don't support dates so converting to millis 
+                    // Scenario should just be theoretical
+                    values = mapValues(property.values, s -> {
+                        Calendar cal = ISO8601.parse(s);
+                        return (cal != null) ? cal.getTimeInMillis() : null;
+                    });
+                    break;
+                case PropertyType.DOUBLE:
+                    values = mapValues(property.values, Double::parseDouble);
+                    break;
+                case PropertyType.LONG:
+                    values = mapValues(property.values, Long::parseLong);
+                    break;
+                case PropertyType.BOOLEAN:
+                    values = mapValues(property.values, Boolean::valueOf);
+                    break;
+                default:
+                    values = property.values;
+            }
+            return values;
+        }
+        
+        @NotNull 
+        private static Object[] mapValues(@NotNull String[] strValues, Function<String, Object> function) {
+            return Arrays.stream(strValues).map(s -> {
```

Comment:
This could be improved by using the isEmptyOrNull in a filter and then mapping the function directly.

Reply:
yes, indeed. thanks.... please take a look if you are ok with the change.

Reply:
ok will merge as discussed